MLP.ipynb

```diff
@@ -1,1146 +1,1188 @@
+    "- **Data Loading**: Loaded test data and moved it to the selected device.\n",
+    "- **Forward Passes**:\n",
+    "  - Passed data through the CNN and converted the output to a numpy array.\n",
+    "  - Passed the numpy array through the fully connected network to get predictions.\n",
+    "- **Accuracy Calculation**: Calculated accuracy by comparing predictions with true labels.\n",
+    "- **Results**:\n",
+    "  - Printed the test accuracy.\n",
+    "  - Generated and printed a classification report using `classification_report` from `sklearn`.\n",
+    "  - Created a confusion matrix and converted it to a DataFrame for better visualization.\n",
+    "- **Wandb Finishing**: Called `wandb.finish()` to complete the logging process."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "q8lI95KDnxBd"
+   },
+   "outputs": [],
+   "source": [
+    "from sklearn.metrics import confusion_matrix, classification_report\n",
+    "import numpy as np\n",
+    "import torch\n",
+    "\n",
+    "def evaluate_model(model, test_loader, device, nn_network, class_names):\n",
+    "    model.eval()  # Set model to evaluation mode\n",
+    "    all_preds = []\n",
+    "    all_labels = []\n",
+    "\n",
+    "    with torch.no_grad():  # Disable gradient calculations for evaluation\n",
+    "        correct = 0\n",
+    "        total = 0\n",
+    "        for x_test, y_test in test_loader:\n",
+    "            # Move data to the selected device\n",
+    "            x_test = x_test.to(device)\n",
+    "            y_test = y_test.to(device)\n",
+    "\n",
+    "            # Forward pass through CNN\n",
+    "            cnn_output_test = model(x_test.type(torch.float32))\n",
+    "\n",
+    "            # Convert to numpy and flatten if needed\n",
+    "            cnn_output_test_np = nn_network.flatten(cnn_output_test.cpu().detach().numpy())\n",
+    "\n",
+    "            # Forward pass through fully connected network\n",
+    "            y_pred = nn_network.feedforward(cnn_output_test_np)\n",
+    "\n",
+    "            # Calculate accuracy\n",
+    "            predicted = np.argmax(y_pred, axis=1)\n",
+    "            correct += (predicted == y_test.cpu().numpy()).sum()\n",
+    "            total += y_test.size(0)\n",
+    "\n",
+    "            all_preds.extend(predicted)\n",
+    "            all_labels.extend(y_test.cpu().numpy())\n",
+    "\n",
+    "    # Calculate accuracy\n",
+    "    accuracy = 100 * correct / total\n",
+    "    print(f\"Accuracy score is: {accuracy:.2f}%\")\n",
+    "    \n",
+    "    return accuracy, all_preds, all_labels\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "accuracy, all_preds, all_labels = evaluate_model(model, test_loader, device, nn_network, class_names)"
```

Comment:
we are returning the accuracy but not using ot, do you want to get use it for print the value outside or you will be using it in hyperparamter?